org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/RefactorProposalUtility.java

```diff
@@ -367,6 +367,11 @@ private static ProposalKindWrapper getExtractVariableAllOccurrenceProposal(CodeA
 		} else {
 			relevance = IProposalRelevance.EXTRACT_LOCAL_ALL;
 		}
+		String declsToFinal = JavaLanguageServerPlugin.getPreferencesManager().getPreferences().getCodeGenerationAddFinalForNewDeclaration();
+		boolean setFinal = false;
+		if ("all".equals(declsToFinal) || "variables".equals(declsToFinal)) {
+			setFinal = true;
```

Comment:
Any reason not to have `boolean setFinal = "all".equals(declsToFinal) || "variables".equals(declsToFinal);` ?

Reply:
Nope, that's much cleaner. Will update, thanks